service/history/stateBuilder.go

```diff
@@ -102,23 +104,26 @@ func (b *stateBuilderImpl) getNewRunTimerTasks() []persistence.Task {
 func (b *stateBuilderImpl) applyEvents(domainID, requestID string, execution shared.WorkflowExecution,
 	history []*shared.HistoryEvent, newRunHistory []*shared.HistoryEvent, eventStoreVersion, newRunEventStoreVersion int32) (*shared.HistoryEvent,
 	*decisionInfo, mutableState, error) {
-	var lastEvent *shared.HistoryEvent
+
+	if len(history) == 0 {
+		return nil, nil, nil, errors.NewInternalFailureError(ErrMessageHistorySizeZero)
```

Comment:
Is this fixing an existing bug or 3dc related? Looks like the behavior changed

Reply:
there is no reason to use the applyEvents function with 0 history